File: Object-C/BlackFireIM/Example/Lib/MSIMSDK-UIKit/ChatDetail/View/Face/MSFaceCollectionCell.m
//
//  MSFaceCollectionCell.m
//  BlackFireIM
//
//  Created by benny wang on 2021/3/23.
//

#import "MSFaceCollectionCell.h"
#import "MSIMSDK-UIKit.h"


@implementation BFFaceCellData
@end

@implementation MSFaceCollectionCell
- (id)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if(self){
        [self setupViews];
        [self defaultLayout];
    }
    return self;
}

- (void)setupViews
{
    _face = [[UIImageView alloc] init];
    _face.contentMode = UIViewContentModeScaleAspectFit;
    [self addSubview:_face];
}

- (void)defaultLayout
{
    CGSize size = self.frame.size;
    _face.frame = CGRectMake(0, 0, size.width, size.height);
}

- (void)setData:(BFFaceCellData * _Nullable)data
{
    self.face.image = [UIImage imageNamed:data.facePath];
    [self defaultLayout];
}


@end

File: Swift/MSIM-Swift/Lib/MSIMSDK-UIKit/ChatDetail/View/Face/MSFaceCollectionCell.swift
//
//  MSFaceCollectionCell.swift
//  MSIM-Swift
//
//  Created by benny wang on 2021/8/3.
//

import UIKit

public class MSFaceCollectionCell: UICollectionViewCell {
    
    var face: UIImageView!
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        face = UIImageView()
        face.contentMode = .scaleAspectFit
        addSubview(face)
        defaultLayout()
    }
    
    private func defaultLayout() {
        face.frame = CGRect(x: 0, y: 0, width: frame.width, height: frame.height)
    }
    
    func setData(data: MSFaceCellData?) {

        face.image = UIImage(named: data?.facePath ?? "")
        defaultLayout()
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}

public class MSFaceCellData: NSObject {
    
    var name: String?
    
    var e_id: String?
    
    var facePath: String?
}
